### ISSUE TITLE
Vim like navigation

### ISSUE THREAD

**Issue by Hainguyen1210** [dailynowco/daily#122]
We are looking for all kinds of feedback and we created a template for almost all of them.
Fill only the relevant section to your issue and delete the other sections, if you can't find a relevant section feel free to go freestyle.

#### Feature Suggestion

It would be faster if I can navigate between posts like in vim.
specifically, I only need four keys h,j,k,l so I think it won't require too much work.
Many thanks

#### Bug Report

Explanation: *Describe the bug as accurate as possible*

Reproduce: *Steps to reproduce the bug, if you know them*

Version: *Daily version (can be found in Chrome/Firefox's extension page)*

Browser version: *Your browser version*

OS: *Your operating system including version*

*Please add screenshots if possible or any other information that can help fixing the bug*

**Comment by idoshamun** [comment]
Hi @Hainguyen1210,

Thank you for suggesting this idea. Let me discuss it internally

**Comment by bpederencino** [comment]
Hi everyone,
 
I would like to point out that I love this feature request. Moreover, it is possible to increase its scope to a keyboard-centric usage of Daily. For instance with keyboard shortcuts to navigate between news cards, access to settings, change layouts and themes, etc.

The first step to implement it could be to add those kinds of key bindings. Then enable the users' possibility to configure them as they want, for example, to have a configuration they are used to as Vim-like or others.

I believe there are two main interests in this kind of feature. The first one is to give an alternative way to use this extension to increase accessibility. And the second one is to have a better integration of Daily in developer workflow and their daily life.

----
### Details
Version: 2.10.4
Browser version: Firefox 70.0.1 (64-bit)
OS: elementary OS 5.1 Hera x86_64

**Comment by idoshamun** [comment]
Hi @bpederencino, I totally agree that we need to embed keyboard shortcuts for the different actions that can be done on Daily. This will make the product much more accessible, as you mentioned, and allow users to quickly navigate. I would love to get a contribution to this one, once I will publish the keybindings.

**Comment by tomsihap** [comment]
An inspiration would be Vimium Chrome extension https://vimium.github.io/ which allows to browse with vim-like shortcuts.

Unfortunately, this extension doesn't work in an homepage like Daily.

**Comment by idoshamun** [comment]
I would love to proceed with this issue and as a starting point to implement navigation through posts as initially suggested here.
As a next step, we can implement more actions with the keyboard.
Does anyone fancy contributing this one? It should be pretty straightforward.

**Comment by sshanzel** [comment]
Let's set the plan and i'll be working on this one soon after!

**Comment by idoshamun** [comment]
Hi @sshanzel,
Thanks for the will to contribute once more and being so active in our community.
I believe a good start should be as @Hainguyen1210 suggested at the beginning.
We'll start by implementing these shortcuts: `h,j,k,l`
These keys should move the focus between the different posts (see image for directions).
For insane mode (list view) `j,k` are the only relevant keys.
Let me know what you think before we start. I will create a technical issue in the `daily-apps` repo, once we confirm the plan.
![hjkl arrow keys](https://user-images.githubusercontent.com/1993245/72619474-09046b80-3946-11ea-8bc2-145f07fdd3af.png)

**Comment by sshanzel** [comment]
Sounds cool! I guess then hit spacebar or enter to open the article right?

**Comment by idoshamun** [comment]
Once you set the focus to the element, the browser will automatically open the article when you hit `enter` as long as the focus is to the anchor (`<a href>`) element. Adding `b` for bookmark is a great idea!

**Comment by tomsihap** [comment]
`/` as a shortcut to the searchbar focus would be nice!

**Comment by sshanzel** [comment]
That is really cool @tomsihap . Thank you again @idoshamun for trusting on another issue! 🤩

**Comment by tomsihap** [comment]
Do you think you can disable the focus to the address bar when you open up a new tab ? So that you just `ctrl+t` and Vim-like navigate right away into Daily without having to press `Tab` or click into the window.

**Comment by sshanzel** [comment]
I will let Ido decide but of course i will make it possible once Ido agrees! 😁

**Comment by idoshamun** [comment]
@tomsihap I believe that this user experience might not be expected by most of the users. Most of the time you open a new tab to navigate somewhere else. You can just hit `ctrl+t` followed by `tab` and then the Vim-like navigation. Hopefully, that makes sense for you. Thank you for sharing your ideas with us :)

**Comment by idoshamun** [comment]
@sshanzel Please comment in the issue so I can assign you:
https://github.com/dailynowco/daily-apps/issues/84

**Comment by idoshamun** [comment]
Thanks to @sshanzel for working on this issue and making it come true.
Hopefully, I will soon release a version with these new key bindings :rocket:

**Comment by sshanzel** [comment]
Thanks a lot @idoshamun for the support all the way! There will be more in the future! It's a pleasure for me to be able to contribute in this wonderful app!!!

**Comment by GiancarloAparicio** [comment]
I was also just looking for something similar on daily.
@idoshamun  seems to me that it would be simpler to integrate daily with vimiumC than to develop all the logic for this.
https://github.com/gdh1995/vimium-c/wiki/Inject-into-other-extensions

**Comment by idoshamun** [comment]
@GiancarloAparicio thanks for the suggestion. I took the time to read the instructions and they require us to modify the permissions of the extension. Modifying permissions is an act that we usually don't take because of our past experience.
We're open to evaluating other alternatives if you're familiar with them. As of daily 3.0, we had to drop our support for the vim-like navigation and we have to implement it again.
